src/WiFi101.h

```diff
@@ -144,10 +146,9 @@ class WiFiClass
 	int hostByName(const char* hostname, IPAddress& result);
 	int hostByName(const String &hostname, IPAddress& result) { return hostByName(hostname.c_str(), result); }
 
-	uint8_t ping(const char* hostname, uint8_t ttl = 128);
-	uint8_t ping(const String &hostname, uint8_t ttl = 128);
-	uint8_t ping(IPAddress host, uint8_t ttl = 128);
-	uint8_t pingGateway();
+	wl_ping_result_t ping(const char* hostname, uint8_t ttl = 128);
```

Comment:
@sandeepmistry , @cmaglie , @facchinm in my oppinion uint8_t is more user friendly. On the other hand wl_ping_result_t tells more about the return type and enables type checking. I'm waiting for your decision.